controller/array_action/array_mediator_svc.py

```diff
@@ -386,13 +384,22 @@ def _delete_volume_by_name(self, volume_name, not_exist_err=True):
 
     def delete_volume(self, volume_id):
         logger.info("Deleting volume with id : {0}".format(volume_id))
-        volume_name = self._get_volume_name_by_wwn(volume_id)
-        fcmap = self._get_fcmap_as_target_if_exists(volume_name)
-        if fcmap:
-            self._stop_and_delete_fcmap(fcmap.id)
-        self._delete_volume_by_name(volume_name)
+        cli_volume = self._get_cli_volume_by_wwn_if_exist(volume_id)
+        if not cli_volume:
+            raise controller_errors.ObjectNotFoundError(volume_id)
+        self._delete_object(cli_volume)
         logger.info("Finished volume deletion. id : {0}".format(volume_id))
 
+    def _safe_delete_fcmaps_as_source(self, object_name):
```

Comment:
could you move this function to be right above `_delete_object`?
it is not used by other methods, so it's best to keep them close

Reply:
done